plugins/inputs/logfile/logfile_test.go

```diff
@@ -358,100 +358,128 @@ func TestLogsFileRemove(t *testing.T) {
+	logFile := NewLogFile()
+	defer logFile.Stop()
+	logFile.Log = TestLogger{t}
+	logFile.FileConfig = []FileConfig{{
 		FilePath:      filepath.Join(filepath.Dir(tmpfile1.Name()), filePrefix+"*"),
 		FromBeginning: true,
 		AutoRemoval:   true,
 	}}
-	tt.FileConfig[0].init()
-	tt.started = true
+	logFile.FileConfig[0].init()
+	logFile.started = true
 
-	lsrcs := tt.FindLogSrc()
-	if len(lsrcs) != 1 {
-		t.Fatalf("%v log src was returned when 1 should be available", len(lsrcs))
-	}
+	logSource, evts := getLogSrc(t, logFile)
+	defer (*logSource).Stop()
 
-	lsrc := lsrcs[0]
-	evts := make(chan logs.LogEvent)
-	lsrc.SetOutput(func(e logs.LogEvent) {
-		if e != nil {
-			evts <- e
-		}
-	})
+	fmt.Println("Fill temp file with sufficient lines to be read.")
+	numLogLinesToWrite := 10000
+	logEntryString := "this is the best log line ever written to a file"
+	writeSomeLines(t, tmpfile1, numLogLinesToWrite, logEntryString)
+	tmpfile1.Close()
 
 	var tmpfile2 *os.File
+	var logSource2 *logs.LogSrc
+	var	evts2 chan logs.LogEvent
 	defer func() {
+		if logSource2 != nil {
+			(*logSource2).Stop()
+		}
 		if tmpfile2 != nil {
+			tmpfile2.Close()
 			os.Remove(tmpfile2.Name())
 		}
 	}()
 
-	wg.Add(1)
+	fmt.Println("Verify every line written to the first temp file is received.")
+	// Do this in a goroutine in case there is a bug and it hangs.
+	readerDone := make(chan bool)
+	// Need to make sure tmpfile2 is created before accessing evts2
+	fileCreatorDone := make(chan bool)
 
 	go func() {
-		defer wg.Done()
-
-		// create a new file matching configured pattern
-		tmpfile2, err = createTempFile("", filePrefix)
-		require.NoError(t, err)
-
-		_, err = tmpfile2.WriteString(logEntryString + "\n")
-		require.NoError(t, err)
-
+		defer close(readerDone)
+		for i := 0; i < numLogLinesToWrite; i++ {
+			logEvent := <- evts
+			require.Equal(t, logEntryString, logEvent.Message())
+			if i != numLogLinesToWrite / 2 {
+				continue
+			}
+			// Halfway through receiving events create a new temp file.
+			// Still expect to recv all events from the first file.
+			// Need to do this in a goroutine since FindLogSrc()
+			// will block until tailer reaches EOF on first temp file.
+			go func() {
+				defer close(fileCreatorDone)
+				tmpfile2, err = createTempFile("", filePrefix)
+				fmt.Printf("Created 2nd temp file, %s\n", tmpfile2.Name())
+				require.NoError(t, err)
+				_, err = tmpfile2.WriteString(logEntryString + "\n")
+				require.NoError(t, err)
+				logSource2, evts2 = getLogSrc(t, logFile)
+			}()
+		}
 	}()
 
-	e := <-evts
-	if e.Message() != logEntryString {
-		t.Errorf("Wrong log found from first file: \n%v\nExpecting:\n%v\n", e.Message(), logEntryString)
-	}
-	defer lsrc.Stop()
-
-	for {
-		lsrcs = tt.FindLogSrc()
-		if len(lsrcs) > 0 {
-			break
-		}
-		time.Sleep(1 * time.Second)
+	fmt.Println("Verify reader completed.")
+	select {
+	case <-readerDone:
+		fmt.Println("Completed before timeout (as expected)")
+	case <-time.After(time.Second * 10):
```

Comment:
Is this 10 seconds meaningful? Ultimately this is for catching a hanging test, so it's not a big deal to leave it like this if this is an arbitrary wait time. I just wanted to see if there was a reason why it was so long.

Reply:
No, if the test passes it doesn't matter how long the timeout is.
It just needs to be large enough that any host running this unit test (e.g. 1 core, 500 MHz) can write/read/recv_events.
I will lower it to 5.